src/common/common-util/src/main/java/ee/ria/xroad/common/conf/globalconf/VersionedConfigurationDirectory.java

```diff
@@ -200,40 +201,52 @@ private void loadSharedParameters(Path instanceDir, Map<String, SharedParameters
-     * @throws Exception if an error occurs while reading parameters
      */
-    public PrivateParameters getPrivate(String instanceId) throws Exception {
+    public Optional<PrivateParameters> findPrivate(String instanceId) {
         String safeInstanceId = escapeInstanceIdentifier(instanceId);
 
-        log.trace("getPrivate(instance = {}, directory = {})", instanceId, safeInstanceId);
+        log.trace("findPrivate(instance = {}, directory = {})", instanceId, safeInstanceId);
 
         PrivateParametersProvider provider = privateParameters.get(safeInstanceId);
-        return provider != null ? provider.getPrivateParameters() : null;
+        return Optional.ofNullable(provider)
+                .map(PrivateParametersProvider::getPrivateParameters);
     }
 
-    public OffsetDateTime getPrivateExpiresOn(String instanceId) {
-        String safeInstanceId = escapeInstanceIdentifier(instanceId);
-        return privateParameters.get(safeInstanceId).getExpiresOn();
+    public boolean isExpired() {
+        OffsetDateTime now = TimeUtils.offsetDateTimeNow();
+        String safeInstanceId = escapeInstanceIdentifier(instanceIdentifier);
+
+        OffsetDateTime privateExpiresOn = privateParameters.get(safeInstanceId).getExpiresOn();
+        if (now.isAfter(privateExpiresOn)) {
+            log.warn("Main privateParameters expired at {}", privateExpiresOn);
+            return true;
+        }
+        OffsetDateTime sharedExpiresOn = sharedParameters.get(safeInstanceId).getExpiresOn();
+        if (now.isAfter(sharedExpiresOn)) {
+            log.warn("Main sharedParameters expired at {}", sharedExpiresOn);
+            return true;
+        }
+        return false;
     }
 
     /**
      * Returns shared parameters for a given instance identifier.
      * @param instanceId the instance identifier
      * @return shared parameters or null, if no shared parameters exist for given instance identifier
      */
-    public SharedParameters getShared(String instanceId) {
+    public Optional<SharedParameters> findShared(String instanceId) {
         String safeInstanceId = escapeInstanceIdentifier(instanceId);
 
-        log.trace("getShared(instance = {}, directory = {})", instanceId, safeInstanceId);
+        log.trace("findShared(instance = {}, directory = {})", instanceId, safeInstanceId);
 
         SharedParametersProvider parameters = sharedParameters.get(safeInstanceId);
         // ignore federated parameters that are expired
         if (parameters != null && parameters.getSharedParameters() != null
                 && (parameters.getSharedParameters().getInstanceIdentifier().equals(instanceIdentifier)
                 || parameters.getExpiresOn().isAfter(TimeUtils.offsetDateTimeNow()))) {
-            return parameters.getSharedParameters();
+            return Optional.of(parameters.getSharedParameters());
         }
-        return null;
+        return Optional.empty();
```

Comment:
Optional map/filter chain would be in style more consistent with `findPrivate()`

Reply:
It would look prettier indeed